pydarn/io/borealis/borealis.py

```diff
@@ -0,0 +1,430 @@
+# Copyright 2019 SuperDARN Canada, University of Saskatchewan
+# Author: Marci Detwiller
+"""
+This file contains classes and functions for 
+converting of Borealis file types. 
+
+Classes
+-------
+BorealisConvert: Writes Borealis SuperDARN files types to
+SuperDARN DARN files with DMap record structure
+
+Exceptions
+----------
+BorealisFileTypeError
+BorealisFieldMissingError
+BorealisExtraFieldError
+BorealisDataFormatTypeError
+BorealisNumberOfRecordsError
+BorealisConversionTypesError
+BorealisConvert2IqdatError
+BorealisConvert2RawacfError
+BorealisRestructureError
+BorealisStructureError
+ConvertFileOverWriteError
+
+Functions
+---------
+read_borealis_file: uses BorealisArrayRead or BorealisSiteRead
+    and returns records or arrays (as requested)
+write_borealis_file: uses BorealisArrayWrite or BorealisSiteWrite
+    to write a file
+return_reader: attempts to read first as BorealisArrayRead and 
+    then as BorealisSiteRead and returns the reader class if one is 
+    successful
+borealis_site_to_array_file: uses BorealisSiteRead and BorealisArrayWrite
+    to convert a site file and write it to an array file.
+borealis_array_to_site_file: uses BorealisArrayRead and BorealisSiteWrite
+    to convert an array file and write it to a site file.
+borealis_write_to_dmap: uses read_borealis_file to read a file 
+    and BorealisConvert to then write a DARN file. 
+bfiq2darniqdat: reads a Borealis bfiq file and converts to DARN iqdat.
+rawacf2darnrawacf: reads a Borealis rawacf file and converts 
+    to DARN rawacf.
```

Comment:
Most of these functions are just wrappers for other functions with one or two inputs specified.  These types of functions should be removed.

Reply:
The intention was to make the borealis IO easy to use by providing some explicitly-named functions for the common use cases. Why would it be better to remove?

Reply:
As a user, I find it more complicated to have two ways to do something, especially when the more general way is very easy to use.  I appreciate the sentiment, but I find the end result to be more confusing.  I think these cases would be better presented as examples in a tutorial.

Reply:
Closing this conversation. @mardet987 and I discussed a resolution that will meet your request and others that I have suggested. 

Duplicate comment.